cognite/client/data_classes/contextualization.py

```diff
@@ -699,10 +699,14 @@ def save_predictions(
             Union[Annotation, AnnotationList]: (suggested) annotation(s) stored in CDF.
 
         """
-        annotations = self._predictions_to_annotations(
-            creating_user=creating_user, creating_app=creating_app, creating_app_version=creating_app_version
+        if self.status == JobStatus.COMPLETED.value:
+            annotations = self._predictions_to_annotations(
+                creating_user=creating_user, creating_app=creating_app, creating_app_version=creating_app_version
+            )
+            if annotations:
+                return self._cognite_client.annotations.suggest(annotations=annotations)
+            raise CogniteException("Extract job does not contain any predictions.")
```

Comment:
Hmm. Should this be an exception? The job completed succesfully - so this would mean every invocation of this method needs to be wrappe din a try-except?

Reply:
Tried to keep the previous behavior, but yes I agree would be better to return empty list or something. Will change it in an upcoming PR :+1: